### PR TITLE
Set mock tl function

### PR DESCRIPTION
**Pull request by ericsciple**

### CODE REVIEW

File: node/mock-run.ts [hunk 1]
```diff
@@ -18,17 +18,28 @@ export class TaskMockRunner {
...
         mockery.registerMock(modName, mod);
     }
-    
-    public run() {
-        mockery.enable({warnOnUnregistered: false});
-        var tlm = require('vsts-task-lib/mock-task');
-        if (this._answers) {
-            tlm.setAnswers(this._answers);
+
+    public run(noMockTask: boolean) {
```
Comment (ericsciple) [inline node/mock-run.ts]: Also I added a bool here whether to register mock-task. What I'm thinking is, the runner is still useful for L1 scenarios where the tl would not be mocked.

**Comment by bryanmacfarlane** [comment]
This makes sense

File: node/mock-run.ts [hunk 2]
```diff
@@ -18,17 +18,28 @@ export class TaskMockRunner {
         this._answers = answers;
     }
 
+    public setFunction(name: string, val: any) {
```
Comment (ericsciple) [inline node/mock-run.ts]: `registerMockExport`

**Comment by ericsciple** [comment]
this is ready for review now

**Pull request merged**